### PR TITLE
feat(auth): add Sign in with Apple directly in Firebase Auth for Android, iOS 13+ and Web

### PR DESCRIPTION
**Pull request by Lyokone**
## Description

We currently rely on [sign_in_with_apple](https://pub.dev/packages/sign_in_with_apple) for Sign In With Apple. We now recommend directly using directly Firebase to connect with Apple.

## Related Issues

Closes https://github.com/firebase/flutterfire/discussions/5560

## Checklist

Before you create this PR confirm that it meets all requirements listed below by checking the relevant checkboxes (`[x]`).
This will ensure a smooth and quick review process. Updating the `pubspec.yaml` and changelogs is not required.

- [ ] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [ ] My PR includes unit or integration tests for *all* changed/updated/fixed behaviors (See [Contributor Guide]).
- [ ] All existing and new tests are passing.
- [ ] I updated/added relevant documentation (doc comments with `///`).
- [ ] The analyzer (`melos run analyze`) does not report any problems on my PR.
- [ ] I read and followed the [Flutter Style Guide].
- [ ] I signed the [CLA].
- [ ] I am willing to follow-up on review comments in a timely manner.

## Breaking Change

Does your PR require plugin users to manually update their apps to accommodate your change?

- [ ] Yes, this is a breaking change.
- [X] No, this is *not* a breaking change.

[issue database]: https://github.com/flutter/flutter/issues
[Contributor Guide]: https://github.com/firebase/flutterfire/blob/master/CONTRIBUTING.md
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[pub versioning philosophy]: https://dart.dev/tools/pub/versioning
[CLA]: https://cla.developers.google.com/

### CODE REVIEW

**Review by russellwheatley — APPROVED** [review]

File: docs/auth/federated-auth.md [hunk 1]
```diff
@@ -171,98 +171,35 @@ For further information, see this [issue](https://github.com/firebase/flutterfir
 
 ## Apple
 
-* {iOS+ and Android}
+* {iOS and macOS}
```
Comment (Salakar) [inline docs/auth/federated-auth.md:174]:
```suggestion
* {iOS+}
```

ios+ afaik covers both

Comment (Lyokone) [reply docs/auth/federated-auth.md]: Oh yeah does make sense 👍

**Review by Salakar — APPROVED** [review]

**Comment by nilsreichardt** [comment]
It also has support for macOS, right?

**Comment by Lyokone** [comment]
It does, but since Firebase iOS SDK does not have official macOS supports, it's not mentioned

**Comment by felixgabler** [comment]
Just out of curiosity, what would go wrong for iOS 11-12?

**Comment by Lyokone** [comment]
@felixgabler Sign In With Apple was introduced in iOS 13. Before you had to open a custom webview with the Apple JS Framework. You can see full [explanations here](https://developer.apple.com/forums/thread/122755?answerId=417003022#417003022).

**Pull request merged**